src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -700,23 +705,31 @@ public function addProductAction(int $orderId, Request $request): Response
         /** @var OrderForViewing $orderForViewing */
         $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
 
-        $products = $orderForViewing->getProducts()->getProducts();
-        $lastProduct = $products[array_key_last($products)];
+        $updatedProducts = $orderForViewing->getProducts()->getProducts();
+        $newProducts = [];
+        for ($i = count($previousProducts); $i < count($updatedProducts); ++$i) {
```

Comment:
@jolelievre am I understanding correctly that you want to return difference between current order products and those added?

Reply:
Not exactly, I want to return the last created products So the added product along with a potential gift added via cart rule